pkg/execution/state/redis_state/lua/includes/gcra.lua

```diff
@@ -1,103 +1,128 @@
--- performs gcra rate limiting for a given key.
---
--- Returns true on success, false if the key has been rate limited.
-local function gcra(key, now_ms, period_ms, limit, burst, enableThrottleFix)
-	-- Calculate the basic variables for GCRA.
-	local cost = 1                            -- everything counts as a single rqeuest
-
-	local emission  = period_ms / math.max(limit, 1)   -- how frequently we can admit new requests
-	local increment = emission * cost         -- this request's time delta
-  -- BUG: The variance is calculated incorrectly. We should use emission instead of period_ms.
-	local variance  = period_ms * (math.max(burst, 1)) -- variance takes into account bursts
-  if enableThrottleFix then
-    -- NOTE: This fixes the delay variation tolerance calculation
-    variance = emission * (math.max(burst, 1))
-  end
-
-	-- fetch the theoretical arrival time for equally spaced requests
-	-- at exactly the rate limit
+-- applyGCRA runs GCRA
+local function applyGCRA(key, now_ms, period_ms, limit, burst, quantity)
+	---@type { allowed: boolean, limit: integer?, retry_after: integer?, reset_after: integer?, remaining: integer? }
+	local result = {}
+
+	-- limit defines the maximum number of requests that can be admitted at once (irrespective of current usage)
+	result["limit"] = burst + 1
```

Comment:
somewhat confusing that `limit` in this result table isn't necessarily the value of the variable `limit` (which is coming from function config, right?), can we name it something different?

Also whatever we name it to, we can make a variable and reuse it on the `dvt` calculation line below `local dvt = emission * (burst + 1)`

Reply:
oh I see, it was matching the `Limit` field in the throttled implementation